tracing-serde/src/lib.rs

```diff
@@ -15,6 +15,36 @@ use tracing_core::{
     span::{Attributes, Id, Record},
 };
 
+/// A bridge between `fmt::Write` and `io::Write`.
+///
+/// This is needed because tracing-subscriber's FormatEvent expects a fmt::Write
+/// while serde_json's Serializer expects an io::Write.
+pub struct WriteAdaptor<'a> {
```

Comment:
Nit: since this is public, a `Debug` impl would be good --- since the `dyn fmt::Write` is not `Debug`, it could just be
```rust
impl fmt::Debug for WriteAdaptor {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.pad("WriteAdaptor { .. }")
    }
}
```